src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,52 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    if (!Animal.alive.includes(this)) {
+      Animal.alive.push(this);
+    }
+  }
+
+  static removeFromAlive(animal) {
+    Animal.alive = Animal.alive.filter((a) => a !== animal);
```

Comment:
Instead of reassigning the filtered array to Animal.alive, you can use Array.prototype.splice() to remove the animal from the original array. This will eliminate the need to create a new array each time an animal dies.

Reply:
Agree, Tnx!